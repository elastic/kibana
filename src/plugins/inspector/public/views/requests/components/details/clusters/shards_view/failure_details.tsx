/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { i18n } from '@kbn/i18n';
import {
  EuiButtonEmpty,
  EuiCodeBlock,
  EuiDescriptionList,
  EuiFlyout,
  EuiFlyoutBody,
  EuiFlyoutFooter,
  EuiFlyoutHeader,
  EuiTitle,
} from '@elastic/eui';
import { getFlattenedObject } from '@kbn/std';

/**
 * Provides pretty formatting of a given key string
 * e.g. formats "this_key.is_nice" to "This key is nice"
 * @param key
 */
export function formatKey(key: string): string {
  const nameCapitalized = key.charAt(0).toUpperCase() + key.slice(1);
  return nameCapitalized.replace(/[\._]/g, ' ');
}
/**
 * Adds a EuiCodeBlock to values of  `script` and `script_stack` key
 * Values of other keys are handled a strings
 * @param value
 * @param key
 */
export function formatValueByKey(value: unknown, key: string): string | JSX.Element {
  if (key === 'script' || key === 'script_stack') {
    const valueScript = Array.isArray(value) ? value.join('\n') : String(value);
    return (
      <EuiCodeBlock language="java" paddingSize="s" isCopyable>
        {valueScript}
      </EuiCodeBlock>
    );
  }

  return String(value);
}

interface Props {
  failure: ShardFailure;
  onClose: () => void;
}

export function FailureDetails({ failure, onClose }: Props) {
  const flattendReason = getFlattenedObject(failure.reason);

  const reasonItems = Object.entries(flattendReason)
    .filter(([key]) => key !== 'type')
    .map(([key, value]) => ({
      title: formatKey(key),
      description: formatValueByKey(value, key),
    }));

  const items = [
    {
      title: i18n.translate('inspector.requests.shardsDetails.shardTitle', {
        defaultMessage: 'Shard',
      }),
      description: failure.shard,
    },
    {
      title: i18n.translate('inspector.requests.shardsDetails.indexTitle', {
        defaultMessage: 'Index',
      }),
      description: failure.index,
    },
    {
      title: i18n.translate('inspector.requests.shardsDetails.reasonTypeTitle', {
        defaultMessage: 'Type',
      }),
      description: failure.reason.type,
    },
    {
      title: i18n.translate('inspector.requests.shardsDetails.nodeTitle', {
        defaultMessage: 'Node',
      }),
      description: failure.node,
    },
    ...reasonItems,
  ];

  return (
    <EuiFlyout onClose={onClose} ownFocus={false}>
      <EuiFlyoutHeader hasBorder>
        <EuiTitle size="s">
          <h1>
            {i18n.translate('inspector.requests.shardsDetails.flyoutTitle', {
              defaultMessage: 'Shard failure',
            })}
          </h1>
        </EuiTitle>
      </EuiFlyoutHeader>

      <EuiFlyoutBody>
        <EuiDescriptionList
          type="responsiveColumn"
          columnWidths={['30%', '70%']}
          listItems={items}
          compressed
        />
      </EuiFlyoutBody>

      <EuiFlyoutFooter>
        <EuiButtonEmpty iconType="cross" onClick={onClose} flush="left">
          {i18n.translate('inspector.requests.shardsDetails.closeButtonLabel', {
            defaultMessage: 'Close',
          })}
        </EuiButtonEmpty>
      </EuiFlyoutFooter>
    </EuiFlyout>
  );
}
