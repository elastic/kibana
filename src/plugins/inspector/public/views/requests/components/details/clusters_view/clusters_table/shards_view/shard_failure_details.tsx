/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { estypes } from '@elastic/elasticsearch';
import { i18n } from '@kbn/i18n';
import { EuiDescriptionList, EuiCodeBlock, EuiText } from '@elastic/eui';
import { getFlattenedObject } from '@kbn/std';

/**
 * Provides pretty formatting of a given key string
 * e.g. formats "this_key.is_nice" to "This key is nice"
 * @param key
 */
export function formatKey(key: string): string {
  const nameCapitalized = key.charAt(0).toUpperCase() + key.slice(1);
  return nameCapitalized.replace(/[\._]/g, ' ');
}
/**
 * Adds a EuiCodeBlock to values of  `script` and `script_stack` key
 * Values of other keys are handled a strings
 * @param value
 * @param key
 */
export function formatValueByKey(value: unknown, key: string): JSX.Element {
  if (key === 'script' || key === 'script_stack') {
    const valueScript = Array.isArray(value) ? value.join('\n') : String(value);
    return (
      <EuiCodeBlock language="java" paddingSize="s" isCopyable>
        {valueScript}
      </EuiCodeBlock>
    );
  }

  return <EuiText size="xs">{String(value)}</EuiText>;
}

interface Props {
  failure: estypes.ShardFailure;
}

export function ShardFailureDetails({ failure }: Props) {
  const flattendReason = getFlattenedObject(failure.reason);

  const reasonItems = Object.entries(flattendReason)
    .filter(([key]) => key !== 'type')
    .map(([key, value]) => ({
      title: formatKey(key),
      description: formatValueByKey(value, key),
    }));

  const items = [
    {
      title: i18n.translate('inspector.requests.clusters.shards.details.nodeLabel', {
        defaultMessage: 'Node',
      }),
      description: formatValueByKey(failure.node, 'node'),
    },
    ...reasonItems,
  ];

  return (
    <EuiText size="xs">
      <EuiDescriptionList
        type="responsiveColumn"
        columnWidths={['30%', '70%']}
        listItems={items}
        compressed
      />
    </EuiText>
  );
}
