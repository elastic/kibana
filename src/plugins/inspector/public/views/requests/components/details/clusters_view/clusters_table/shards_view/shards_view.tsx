/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { estypes } from '@elastic/elasticsearch';
import { i18n } from '@kbn/i18n';
import { EuiFlexGroup, EuiFlexItem, EuiTitle } from '@elastic/eui';
import { OpenShardFailureFlyoutButton } from './open_shard_failure_flyout_button';

interface Props {
  failures: estypes.ShardFailure[];
  shardStats?: estypes.ShardStatistics;
}

export function ShardsView({ failures, shardStats }: Props) {
  return !shardStats && failures.length === 0 ? null : (
    <>
      <EuiFlexGroup justifyContent="spaceBetween">
        <EuiFlexItem grow={false}>
          <EuiTitle size="xxxs">
            <h4>
              {i18n.translate('inspector.requests.clusters.shards.shardsTitle', {
                defaultMessage: 'Shards',
              })}
            </h4>
          </EuiTitle>
        </EuiFlexItem>

        <EuiFlexItem grow={false}>
          <OpenShardFailureFlyoutButton failures={failures} />
        </EuiFlexItem>
      </EuiFlexGroup>

      {shardStats ? (
        <EuiFlexGroup justifyContent="spaceBetween">
          <EuiFlexItem grow={false}>
            {i18n.translate('inspector.requests.clusters.shards.totalShardsLabel', {
              defaultMessage: '{total} total shards',
              values: { total: shardStats.total },
            })}
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            {i18n.translate('inspector.requests.clusters.shards.successfulShardsLabel', {
              defaultMessage: '{successful} of {total} successful',
              values: {
                successful: shardStats.successful,
                total: shardStats.total,
              },
            })}
          </EuiFlexItem>
        </EuiFlexGroup>
      ) : null}
    </>
  );
}
