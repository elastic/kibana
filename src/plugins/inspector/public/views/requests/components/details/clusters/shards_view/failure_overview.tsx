/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { useState } from 'react';
import { i18n } from '@kbn/i18n';
import { EuiButtonEmpty, EuiDescriptionList, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
import { FailureDetails } from './failure_details';

interface Props {
  failure: ShardFailure;
}

export function FailureOverview({ failure }: Props) {
  const [showDetails, setShowDetails] = useState(false);

  const items = [
    {
      title: i18n.translate('inspector.requests.shardsDetails.shardTitle', {
        defaultMessage: 'Shard',
      }),
      description: failure.shard,
    },
    {
      title: i18n.translate('inspector.requests.shardsDetails.indexTitle', {
        defaultMessage: 'Index',
      }),
      description: failure.index ?? '',
    },
    {
      title: i18n.translate('inspector.requests.shardsDetails.reasonTypeTitle', {
        defaultMessage: 'Type',
      }),
      description: failure.reason.type,
    },
  ];

  return (
    <>
      <EuiFlexGroup justifyContent="spaceBetween">
        <EuiFlexItem grow={false}>
          <EuiDescriptionList type="inline" listItems={items} compressed />
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiButtonEmpty
            flush="right"
            onClick={() => {
              setShowDetails(true);
            }}
            size="xs"
          >
            {i18n.translate('inspector.requests.shardsDetails.viewDetailsLabel', {
              defaultMessage: 'Details',
            })}
          </EuiButtonEmpty>
        </EuiFlexItem>
      </EuiFlexGroup>
      {showDetails ? (
        <FailureDetails
          failure={failure}
          onClose={() => {
            setShowDetails(false);
          }}
        />
      ) : null}
    </>
  );
}
