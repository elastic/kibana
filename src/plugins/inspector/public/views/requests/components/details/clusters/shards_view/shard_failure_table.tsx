/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { useState, ReactNode } from 'react';
import { estypes } from '@elastic/elasticsearch';
import { i18n } from '@kbn/i18n';
import { EuiBasicTable, EuiButtonIcon, EuiText } from '@elastic/eui';
import { ShardFailureDetails } from './shard_failure_details';

interface Props {
  failures: estypes.ShardFailure[];
}

export function ShardFailureTable({ failures }: Props) {
  const [expandedRows, setExpandedRows] = useState<Record<number, ReactNode>>({});

  const toggleDetails = (shard: number) => {
    const nextExpandedRows = { ...expandedRows };
    if (shard in nextExpandedRows) {
      delete nextExpandedRows[shard];
    } else {
      const shardFailure = failures.find(failure => shard === failure.shard);
      nextExpandedRows[shard] = shardFailure ? <ShardFailureDetails failure={shardFailure} /> : null;
    }
    setExpandedRows(nextExpandedRows);
  };

  const columns = [
    {
      field: 'shard',
      name: i18n.translate('inspector.requests.clusters.shards.table.shardLabel', {
        defaultMessage: 'Shard',
      }),
      render: (shard: string) => {
        return (
          <>
            <EuiButtonIcon
              onClick={() => toggleDetails(shard)}
              aria-label={
                shard in expandedRows
                  ? i18n.translate('inspector.requests.clusters.shards.table.collapseRow', {
                      defaultMessage: 'Collapse table row to hide shard details',
                    })
                  : i18n.translate('inspector.requests.clusters.shards.table.expandRow', {
                      defaultMessage: 'Expand table row to view shard details',
                    })
              }
              iconType={shard in expandedRows ? 'arrowDown' : 'arrowRight'}
            />
            <EuiText size="xs" color="subdued">
              {shard}
            </EuiText>
          </>
        );
      },
      width: '20%',
    },
    {
      field: 'index',
      name: i18n.translate('inspector.requests.clusters.shards.table.indexLabel', {
        defaultMessage: 'Index',
      }),
      render: (index?: string) =>
        index ? (
          <EuiText size="xs" color="subdued">
            {index}
          </EuiText>
        ) : null,
    },
    {
      field: 'failureType',
      name: i18n.translate('inspector.requests.clusters.shards.table.failureTypeLabel', {
        defaultMessage: 'Failure type',
      }),
      render: (failureType: string) => (
        <EuiText size="xs" color="subdued">
          {failureType}
        </EuiText>
      ),
    },
  ];

  return (
    <EuiBasicTable
      items={failures.map(({ shard, index, reason }) => {
        return {
          shard,
          index,
          failureType: reason.type,
        };
      })}
      isExpandable={true}
      itemIdToExpandedRowMap={expandedRows}
      itemId="shard"
      columns={columns}
    />
  );
}
