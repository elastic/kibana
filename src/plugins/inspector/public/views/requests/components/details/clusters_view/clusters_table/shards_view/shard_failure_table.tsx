/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { useState, ReactNode } from 'react';
import { estypes } from '@elastic/elasticsearch';
import { i18n } from '@kbn/i18n';
import { EuiBasicTable, type EuiBasicTableColumn, EuiButtonIcon, EuiText } from '@elastic/eui';
import { ShardFailureDetails } from './shard_failure_details';

function getRowId(failure: estypes.ShardFailure) {
  return `${failure.shard}${failure.index}`;
}

interface ShardRow {
  rowId: string;
  shard: number;
  index?: string;
  failureType: string;
}

interface Props {
  failures: estypes.ShardFailure[];
}

export function ShardFailureTable({ failures }: Props) {
  const [expandedRows, setExpandedRows] = useState<Record<string, ReactNode>>({});

  const toggleDetails = (rowId: string) => {
    const nextExpandedRows = { ...expandedRows };
    if (rowId in nextExpandedRows) {
      delete nextExpandedRows[rowId];
    } else {
      const shardFailure = failures.find((failure) => rowId === getRowId(failure));
      nextExpandedRows[rowId] = shardFailure ? (
        <ShardFailureDetails failure={shardFailure} />
      ) : null;
    }
    setExpandedRows(nextExpandedRows);
  };

  const columns: Array<EuiBasicTableColumn<ShardRow>> = [
    {
      field: 'shard',
      name: i18n.translate('inspector.requests.clusters.shards.table.shardLabel', {
        defaultMessage: 'Shard',
      }),
      render: (shard: number, item: ShardRow) => {
        return (
          <>
            <EuiButtonIcon
              onClick={() => toggleDetails(item.rowId)}
              aria-label={
                item.rowId in expandedRows
                  ? i18n.translate('inspector.requests.clusters.shards.table.collapseRow', {
                      defaultMessage: 'Collapse table row to hide shard details',
                    })
                  : i18n.translate('inspector.requests.clusters.shards.table.expandRow', {
                      defaultMessage: 'Expand table row to view shard details',
                    })
              }
              iconType={item.rowId in expandedRows ? 'arrowDown' : 'arrowRight'}
            />
            <EuiText size="xs" color="subdued">
              {shard}
            </EuiText>
          </>
        );
      },
      width: '20%',
    },
    {
      field: 'index',
      name: i18n.translate('inspector.requests.clusters.shards.table.indexLabel', {
        defaultMessage: 'Index',
      }),
      render: (index?: string) =>
        index ? (
          <EuiText size="xs" color="subdued">
            {index}
          </EuiText>
        ) : null,
    },
    {
      field: 'failureType',
      name: i18n.translate('inspector.requests.clusters.shards.table.failureTypeLabel', {
        defaultMessage: 'Failure type',
      }),
      render: (failureType: string) => (
        <EuiText size="xs" color="subdued">
          {failureType}
        </EuiText>
      ),
    },
  ];

  return (
    <EuiBasicTable
      items={failures.map((failure) => {
        return {
          rowId: getRowId(failure),
          shard: failure.shard,
          index: failure.index,
          failureType: failure.reason.type,
        };
      })}
      isExpandable={true}
      itemIdToExpandedRowMap={expandedRows}
      itemId="rowId"
      columns={columns}
    />
  );
}
