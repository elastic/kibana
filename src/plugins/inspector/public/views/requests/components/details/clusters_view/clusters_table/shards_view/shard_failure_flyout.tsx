/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { estypes } from '@elastic/elasticsearch';
import { i18n } from '@kbn/i18n';
import {
  EuiButtonIcon,
  EuiButtonEmpty,
  EuiFlyout,
  EuiFlyoutBody,
  EuiFlyoutFooter,
  EuiFlyoutHeader,
  EuiTitle,
} from '@elastic/eui';
import { ShardFailureTable } from './shard_failure_table';

interface Props {
  failures: estypes.ShardFailure[];
  onClose: () => void;
}

export function ShardFailureFlyout({ failures, onClose }: Props) {
  return (
    <EuiFlyout onClose={onClose} ownFocus={false} hideCloseButton={true}>
      <EuiFlyoutHeader hasBorder>
        <EuiTitle size="s">
          <h1>
            <EuiButtonIcon iconType="sortLeft" onClick={onClose} />
            {i18n.translate('inspector.requests.clusters.shards.flyoutTitle', {
              defaultMessage:
                '{failedShardCount} failed {failedShardCount, plural, one {shard} other {shards}}',
              values: { failedShardCount: failures.length },
            })}
          </h1>
        </EuiTitle>
      </EuiFlyoutHeader>

      <EuiFlyoutBody>
        <ShardFailureTable failures={failures} />
      </EuiFlyoutBody>

      <EuiFlyoutFooter>
        <EuiButtonEmpty iconType="sortLeft" onClick={onClose} flush="left">
          {i18n.translate('inspector.requests.clusters.shards.backButtonLabel', {
            defaultMessage: 'Back',
          })}
        </EuiButtonEmpty>
      </EuiFlyoutFooter>
    </EuiFlyout>
  );
}
