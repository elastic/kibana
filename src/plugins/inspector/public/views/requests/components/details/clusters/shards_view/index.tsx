/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { useState } from 'react';
import { i18n } from '@kbn/i18n';
import { EuiButtonEmpty, EuiFlexGroup, EuiFlexItem, EuiTitle } from '@elastic/eui';
import { ShardFailureFlyout } from './shard_failure_flyout';

interface Props {
  failures: ShardFailure[];
  shardsDetails?: ShardStatistics;
}

export function ShardsView({ failures, shardsDetails }: Props) {
  const [showFailures, setShowFailures] = useState(false);

  return !shardsDetails || failures.length === 0 ? null : (
    <>
      <EuiFlexGroup justifyContent="spaceBetween">
        <EuiFlexItem grow={false}>
          <EuiTitle size="xxxs">
            <h4>
              {i18n.translate('inspector.requests.shards.shardsTitle', {
                defaultMessage: 'Shards',
              })}
            </h4>
          </EuiTitle>
        </EuiFlexItem>

        {failures.length ? (
          <EuiFlexItem grow={false}>
            <EuiButtonEmpty
              flush="left"
              onClick={() => {
                setShowFailures(!showFailures);
              }}
              size="xs"
            >
              {i18n.translate('inspector.requests.clusters.shards.hideFailuresLabel', {
                defaultMessage:
                  'View {failedShardCount} failed {failedShardCount, plural, one {shard} other {shards}}',
                values: { failedShardCount: failures.length },
              })}
            </EuiButtonEmpty>
          </EuiFlexItem>
        ) : null}
      </EuiFlexGroup>

      {shardsDetails ? (
        <EuiFlexGroup justifyContent="spaceBetween">
          <EuiFlexItem grow={false}>
            {i18n.translate('inspector.requests.clusters.shards.totalShardsLabel', {
              defaultMessage: '{total} total shards',
              values: { total: shardsDetails.total },
            })}
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            {i18n.translate('inspector.requests.clusters.shards.successfulShardsLabel', {
              defaultMessage: '{successful} of {total} successful',
              values: {
                successful: shardsDetails.successful,
                total: shardsDetails.total,
              },
            })}
          </EuiFlexItem>
        </EuiFlexGroup>
      ) : null}

      {showFailures ? (
        <ShardFailureFlyout
          failures={failures}
          onClose={() => {
            setShowFailures(false);
          }}
        />
      ) : null}
    </>
  );
}
