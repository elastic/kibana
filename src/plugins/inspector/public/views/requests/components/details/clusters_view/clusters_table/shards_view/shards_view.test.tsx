/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { estypes } from '@elastic/elasticsearch';
import { shallow } from 'enzyme';
import { ShardsView } from './shards_view';

describe('render', () => {
  test('should render with no failures', () => {
    const shardStats = {
      total: 2,
      successful: 2,
      skipped: 0,
      failed: 0,
    };
    const wrapper = shallow(<ShardsView failures={[]} shardStats={shardStats} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render with failures', () => {
    const shardStats = {
      total: 2,
      successful: 1,
      skipped: 0,
      failed: 1,
    };
    const wrapper = shallow(
      <ShardsView failures={[{} as unknown as estypes.ShardFailure]} shardStats={shardStats} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should not render when no shard details are provided', () => {
    const wrapper = shallow(<ShardsView failures={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
