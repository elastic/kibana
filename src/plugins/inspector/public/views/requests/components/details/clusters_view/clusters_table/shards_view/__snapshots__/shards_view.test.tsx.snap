// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should not render when no shard details are provided 1`] = `""`;

exports[`render should render with failures 1`] = `
<Fragment>
  <EuiFlexGroup
    justifyContent="spaceBetween"
  >
    <EuiFlexItem
      grow={false}
    >
      <EuiTitle
        size="xxxs"
      >
        <h4>
          Shards
        </h4>
      </EuiTitle>
    </EuiFlexItem>
    <EuiFlexItem
      grow={false}
    >
      <OpenShardFailureFlyoutButton
        failures={
          Array [
            Object {},
          ]
        }
      />
    </EuiFlexItem>
  </EuiFlexGroup>
  <EuiFlexGroup
    justifyContent="spaceBetween"
  >
    <EuiFlexItem
      grow={false}
    >
      2 total shards
    </EuiFlexItem>
    <EuiFlexItem
      grow={false}
    >
      1 of 2 successful
    </EuiFlexItem>
  </EuiFlexGroup>
</Fragment>
`;

exports[`render should render with no failures 1`] = `
<Fragment>
  <EuiFlexGroup
    justifyContent="spaceBetween"
  >
    <EuiFlexItem
      grow={false}
    >
      <EuiTitle
        size="xxxs"
      >
        <h4>
          Shards
        </h4>
      </EuiTitle>
    </EuiFlexItem>
    <EuiFlexItem
      grow={false}
    >
      <OpenShardFailureFlyoutButton
        failures={Array []}
      />
    </EuiFlexItem>
  </EuiFlexGroup>
  <EuiFlexGroup
    justifyContent="spaceBetween"
  >
    <EuiFlexItem
      grow={false}
    >
      2 total shards
    </EuiFlexItem>
    <EuiFlexItem
      grow={false}
    >
      2 of 2 successful
    </EuiFlexItem>
  </EuiFlexGroup>
</Fragment>
`;
